### PR TITLE
FreeBSD: enable spa_thread

### PR DESCRIPTION
**Pull request by kev009**
### Motivation and Context

Retry https://github.com/openzfs/zfs/pull/10749
This seems to be fixed via https://github.com/openzfs/zfs/commit/ca14e08cbff36cadd26928cb01222707930973cb

### Description

Avoid a cosmetic regression when downstreaming.
Bind each pool to its own thread, put the pool
name in the proc title, etc.

Signed-off-by: Matt Macy <mmacy@FreeBSD.org>
Signed-off-by: Kevin Bowling <kbowling@FreeBSD.org>

### How Has This Been Tested?

ozfs test suite ran by @freqlabs, looking to PR builds on linux

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by mmatuska** [comment]
Could you please fill up the "Types of changes" and "Checklist"?

**Comment by freqlabs** [comment]
I wonder why ca14e08 fixes this? This is likely still broken on platforms that !HAVE_LARGE_STACKS.

**Comment by freqlabs** [comment]
Dunno where the FreeBSD test bots are.

**Comment by kev009** [comment]
> I wonder why [ca14e08](https://github.com/openzfs/zfs/commit/ca14e08cbff36cadd26928cb01222707930973cb) fixes this? This is likely still broken on platforms that !HAVE_LARGE_STACKS.

I don't think the zpool command was ever meant to run in userspace with the alternate implementation, which is likely where the fix comes from.  There is no reason to, the smallest stack on Linux would be 128k with musl, 2M freebsd libc, 8M with glibc.

MIPS will be deleted in FreeBSD main.  I have a change to increase the KSTACK_PAGES on armv7 https://reviews.freebsd.org/D31244.  So my suggestion is we don't merge this to 2.1.  To be frank though, I am skeptical zfs would work ok with the current situation on arm and somebody that cares needs to run a test suite before and after.  I have no armv7 HW nor any inclination to prop it up.

**Comment by kev009** [comment]
It looks like the arm maintainers are on board with the change.  We already have a sufficient warning here https://github.com/openzfs/zfs/blob/60ffc1c460e4cdf3c3ca12c8840fd0675a98ed0d/module/os/freebsd/zfs/kmod_core.c#L239

**Comment by behlendorf** [comment]
@kev009 we should have sorted out the FreeBSD bots, so they should run the next time you force update this PR.

**Review by freqlabs — COMMENTED** [review]
I want to go through and make sure that rebasing didn't miss anything (or the original version for that matter) when I have time.

File: module/os/freebsd/zfs/spa_os.c [hunk 1]
```diff
@@ -73,6 +73,30 @@
 #include "zfs_prop.h"
 #include "zfs_comutil.h"
 
+
+int
+newproc(void (*pc)(void *), void *arg, id_t cid, int pri,
+    void **ct, pid_t pid)
+{
+	spa_t *spa = (spa_t *)arg;	/* XXX */
```
[the remark this answers is not in the document]
Comment (kev009) [reply module/os/freebsd/zfs/spa_os.c:81]: @amotin gave me some advice in private, I will submit some cleanups to the patch when I have the energy to do so.

**Comment by behlendorf** [comment]
Hey folks, to help reduce the number of dormant PRs I'd like to see if we either move this forward or close it out for now.  If you can rebase this PR and address the review feedback it seems like something we could wrap up.  It'd be nice to also sort this out for Linux but I don't think that needs to be a blocker.

**Comment by kev009** [comment]
> Hey folks, to help reduce the number of dormant PRs I'd like to see if we either move this forward or close it out for now. If you can rebase this PR and address the review feedback it seems like something we could wrap up. It'd be nice to also sort this out for Linux but I don't think that needs to be a blocker.

@behlendorf @amotin Closing the PR as I don't have time to work on this with a new job.

**Pull request closed without merging**